pyccel/epyccel.py

```diff
@@ -181,28 +237,47 @@ def epyccel( inputs, **kwargs ):
 
         # Master process calls epyccel
         if comm.rank == root:
-            fmod      = epyccel_seq( inputs, **kwargs )
-            fmod_path = os.path.abspath(fmod.__file__)
-            fmod_name = fmod.__name__
+            try:
+                mod, fun = epyccel_seq( python_function_or_module, **kwargs )
+                mod_path = os.path.abspath(mod.__file__)
+                mod_name = mod.__name__
+                fun_name = python_function_or_module.__name__ if fun else None
+                success  = True
+            except:
```

Comment:
Codacy is complaining about this as it is very bad practice. Is this supposed to catch everything (even typos in the python code) and broadcast it or is this supposed to catch something more specific?

Reply:
Yes, it should catch anything! I just need to broadcast the exception information, whatever it is.

Reply:
Then I think we should add a comment explaining why this is an exception and deactivate the pylint error for this line

Reply:
Should I write `except Exception`?

Reply:
maybe. That doesn't catch everything though. I think system.exit() isn't caught

Reply:
I would just do something like:
```
# error handling carried out after broadcast to prevent deadlocks
except: # pylint: disable=bare-except
```